server/routes/creatingReports/reportMayAlreadyExist.test.ts

```diff
@@ -0,0 +1,143 @@
+import moment from 'moment'
+import request from 'supertest'
+import { ReportService, LocationService, OffenderService } from '../../services'
+import { appWithAllRoutes } from '../__test/appSetup'
+
+jest.mock('../../services/offenderService')
+jest.mock('../../services/reportService')
+jest.mock('../../services/locationService')
+
+const offenderService = new OffenderService(null) as jest.Mocked<OffenderService>
+const locationService = new LocationService(null) as jest.Mocked<LocationService>
+const reportService = new ReportService(null, null, null, null, null, null) as jest.Mocked<ReportService>
+
+let app
+
+beforeEach(() => {
+  app = appWithAllRoutes({ reportService, locationService, offenderService })
+  offenderService.getOffenderDetails.mockResolvedValue({})
+  reportService.getReportInProgress.mockResolvedValue({ id: null, incidentdate: null })
+  locationService.getLocation.mockResolvedValue({})
+  reportService.getReportsByDate.mockResolvedValue([])
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('/report-may-already-exist', () => {
+  describe('GET', () => {
+    it('should get offender name', () => {
+      return request(app)
+        .get('/report/2/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(() => {
+          expect(offenderService.getOffenderDetails).toBeCalledTimes(1)
+          expect(offenderService.getOffenderDetails).toBeCalledWith('user1-system-token', 2)
+        })
+    })
+
+    it('should get in-progress report', () => {
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(() => {
+          expect(reportService.getReportInProgress).toBeCalledTimes(1)
+          expect(reportService.getReportInProgress).toBeCalledWith(3)
+        })
+    })
+
+    it('should redirect if no in-progress report', () => {
+      reportService.getReportInProgress.mockResolvedValue(null)
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect(302)
+        .expect('Location', '/report/3/report-has-been-deleted')
+    })
+
+    it('should get reports that have same incident date as the new in-progress report', () => {
+      reportService.getReportInProgress.mockResolvedValue({ id: 15, incidentdate: '10/07/2021' })
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(() => {
+          expect(reportService.getReportInProgress).toBeCalledTimes(1)
```

Comment:
typo? should this be checking `getReportsByDate` ? 

There's not really any benefit doing these `toBeCalledTimes` as `toBeCalledWith` will fail when never called with a nice message. Would suggest deleting them..

Reply:
deleted